lib/cogctl/actions/roles/grant.ex

```diff
@@ -1,54 +1,32 @@
 defmodule Cogctl.Actions.Roles.Grant do
   use Cogctl.Action, "roles grant"
 
+  alias CogApi.HTTP.Client
+
   def option_spec do
     [{:role, :undefined, :undefined, {:string, :undefined}, 'Role name (required)'},
-     {:user_to_grant, :undefined, 'user', {:string, :undefined}, 'Username of user to grant role'},
-     {:group_to_grant, :undefined, 'group', {:string, :undefined}, 'Name of group to grant role'}]
-  end
-
-  def run(options, _args, _config, endpoint) do
-    role = :proplists.get_value(:role, options)
-    user_to_grant = :proplists.get_value(:user_to_grant, options)
-    group_to_grant = :proplists.get_value(:group_to_grant, options)
-
-    with_authentication(endpoint,
-                        &do_grant(&1, role, user_to_grant, group_to_grant))
-  end
-
-  defp do_grant(_endpoint, :undefined, _user_to_grant, _group_to_grant) do
-    display_arguments_error
+     {:group, :undefined, 'group', {:string, :undefined}, 'Group name (required)'}]
   end
 
-  defp do_grant(_endpoint, _role, :undefined, :undefined) do
-    display_arguments_error
+  def run(options, _args, _config, %{token: nil}=endpoint) do
+    with_authentication(endpoint, &run(options, nil, nil, &1))
   end
 
-  defp do_grant(endpoint, role, user_to_grant, :undefined) do
-    case CogApi.HTTP.Internal.role_grant(endpoint, role, "users", user_to_grant) do
-      {:ok, _resp} ->
-        display_output("Granted #{role} to #{user_to_grant}")
-      {:error, error} ->
-        display_error(error["errors"])
-    end
+  def run(options, _args, _config, endpoint) do
+    group = Client.group_find(endpoint, name: :proplists.get_value(:group, options))
+    role = Client.role_show(endpoint, %{name: :proplists.get_value(:role, options)})
+    do_grant(endpoint, role, group)
```

Comment:
That's fair, I may have gotten a little too clever for myself here. I was trying to avoid writing more explicit error checks for both group and role. I'll rework this for clarity.